pkg/ast/spl/spl.peg

```diff
@@ -505,6 +442,170 @@ Start <- SPACE? initialSearch:(InitialSearchBlock) filterBlocks:(FilterBlock)* q
+        },
+    }, nil
+}
+
+GenTimesOptionStart <- "start" EQUAL timeStamp:(GenTimestamp) {
+    return &GenTimesOptionArgs{
+        argOption: "start",
+        genTimesOption: &structs.GenTimes{
+            StartTime: uint64(timeStamp.(int64)),
+        },
+    }, nil
+}
+
+GenTimesOptionIncrement <- "increment" EQUAL intStr:IntegerAsString unitStr:("s" / "m" / "d" / "h")? {
+    spanNum, err := strconv.ParseInt(intStr.(string), 10, 64)
+    if err != nil {
+        return "", fmt.Errorf("Spl peg: Error while converting the integer: %v", err)
+    }
+    unit, err := transferUint8ToString(unitStr)
+    if err != nil {
+        return nil, fmt.Errorf("Spl peg: Unit specified is not string, err: %v", err)
+    }
+
+    timeUnit := utils.TMDay
+    if unit == "s" {
+        timeUnit = utils.TMSecond
+    } else if unit == "m" {
+        timeUnit = utils.TMMinute
+    } else if unit == "h" {
+        timeUnit = utils.TMHour
+    }
+
+    return &GenTimesOptionArgs{
+        argOption: "increment",
+        genTimesOption: &structs.GenTimes{
+            Interval: &structs.SpanLength{
+                Num: int(spanNum),
+                TimeScalr: timeUnit,
+            },
+        },
+    }, nil
+}
+
+GenTimesOption <- genTimesOption:(GenTimesOptionEnd / GenTimesOptionStart / GenTimesOptionIncrement) {
+    return genTimesOption, nil
+}
+
+GenTimesOptionList <- first:GenTimesOption rest:(SPACE GenTimesOption)* {
+    restSlice := rest.([]any)
+    optionWasSpecified := make(map[string]bool)
+
+    numOptions := 1 + len(restSlice)
+
+    genTimeOption, err := initializeGenTimesOption()
+    if err != nil {
+        return nil, fmt.Errorf("Spl peg: Error while initializing the GenTimesOption: %v", err)
+    }
+
+    for i := 0; i < numOptions; i++ {
+        var genTimeOptArg *GenTimesOptionArgs
+        if i == 0 {
+            genTimeOptArg = first.(*GenTimesOptionArgs)
+        } else {
+            separator := restSlice[i - 1].([]any)
+            genTimeOptArg = separator[1].(*GenTimesOptionArgs)
+        }
+
+        argOption := genTimeOptArg.argOption
+        val, exist := optionWasSpecified[argOption]
+        if exist && val {
+            return nil, fmt.Errorf("Invalid Syntax, option: %v is already specified", argOption)
+        }
+        optionWasSpecified[argOption] = true
+
+        switch argOption {
+            case "start":
+                genTimeOption.StartTime = genTimeOptArg.genTimesOption.StartTime
+            case "end":
+                genTimeOption.EndTime = genTimeOptArg.genTimesOption.EndTime
+            case "increment":
+                genTimeOption.Interval = genTimeOptArg.genTimesOption.Interval
+            default:
+                return nil, fmt.Errorf("Not a Valid GenTimes Command Option: %v", argOption)
+        }
+    }
+
+    startSpecified, exist := optionWasSpecified["start"]
+    if !exist || !startSpecified {
+        return nil, fmt.Errorf("Invalid Syntax, start time is not specified")
+    }
```

Comment:
Just checking for `exist` should be fine, right? As seems like the values if present in `optionWasSpecified` will always be `true`.  Also, if that is the case you should change this `optionWasSpecified` type to `map[string]struct{}` instead of `map[string]bool`.

Reply:
addressed in the latest commit.